src/iosource/Packet.h

```diff
@@ -221,11 +221,45 @@ class Packet {
 	// These are fields passed between various packet analyzers. They're best
 	// stored with the packet so they stay available as the packet is passed
 	// around.
+
+	/**
+	 * The stack of encapsulations this packet belongs to, if any. This is
+	 * used by the tunnel analyzers to keep track of the encapsulations as
+	 * processing occurs.
+	 */
 	EncapsulationStack* encap = nullptr;
+
+	/**
+	 * The IP header for this packet. This is filled in by the IP analyzer
+	 * during processing if the packet contains an IP header.
+	 */
 	IP_Hdr* ip_hdr = nullptr;
+
+	/**
+	 * The protocol of the packet. This is used by the tunnel analyzers to
+	 * pass outer protocol from one level to the next.
+	 */
 	int proto = -1;
+
+	/**
+	 * If the packet contains a tunnel, this field will be filled in with
+	 * the type of tunnel. It is used to pass the tunnel type between the
+	 * packet analyzers during analysis.
+	 */
 	BifEnum::Tunnel::Type tunnel_type = BifEnum::Tunnel::IP;
+
+	/**
+	 * If the packet contains a GRE tunnel, this fieidl will contain the
```

Comment:
```suggestion
	 * If the packet contains a GRE tunnel, this field will contain the
```

Reply:
Yikes, what were my fingers doing with the keys on that one?